snuba/subscriptions/data.py

```diff
@@ -175,68 +188,98 @@ def to_dict(self) -> Mapping[str, Any]:
+                Column(None, None, required_timestamp_column),
+                Literal(None, (timestamp - self.time_window)),
+            ),
+            binary_condition(
+                ConditionFunctions.LT,
+                Column(None, None, required_timestamp_column),
+                Literal(None, timestamp),
+            ),
+        ]
 
-            conditions_to_add: List[Expression] = [
-                binary_condition(
-                    ConditionFunctions.EQ,
-                    Column(None, None, "project_id"),
-                    Literal(None, self.project_id),
-                ),
-                binary_condition(
-                    ConditionFunctions.GTE,
-                    Column(None, None, required_timestamp_column),
-                    Literal(None, (timestamp - self.time_window)),
-                ),
+        if offset is not None:
+            conditions_to_add.append(
                 binary_condition(
-                    ConditionFunctions.LT,
-                    Column(None, None, required_timestamp_column),
-                    Literal(None, timestamp),
-                ),
-            ]
-
-            if offset is not None:
-                conditions_to_add.append(
-                    binary_condition(
-                        ConditionFunctions.LTE,
-                        FunctionCall(
-                            None,
-                            "ifnull",
-                            (Column(None, None, "offset"), Literal(None, 0)),
-                        ),
-                        Literal(None, offset),
-                    )
+                    ConditionFunctions.LTE,
+                    FunctionCall(
+                        None,
+                        "ifnull",
+                        (Column(None, None, "offset"), Literal(None, 0)),
+                    ),
+                    Literal(None, offset),
                 )
+            )
 
-            new_condition = combine_and_conditions(conditions_to_add)
-            condition = query.get_condition()
-            if condition:
-                new_condition = binary_condition(
-                    BooleanFunctions.AND, condition, new_condition
-                )
+        new_condition = combine_and_conditions(conditions_to_add)
+        condition = query.get_condition()
+        if condition:
+            new_condition = binary_condition(
+                BooleanFunctions.AND, condition, new_condition
+            )
+
+        query.set_ast_condition(new_condition)
+
+    def validate_subscription(
+        self, query: Union[CompositeQuery[Entity], Query]
+    ) -> None:
+        # TODO: Support composite queries with multiple entities.
+        from_clause = query.get_from_clause()
+        if not isinstance(from_clause, Entity):
+            raise InvalidSubscriptionError("Only simple queries are supported")
+        entity = get_entity(from_clause.key)
+
+        SubscriptionAllowedClausesValidator().validate(query)
```

Comment:
Validators felt like the right place to put this logic, but I'm open to suggestions.